### DISCUSSION TITLE
Seq tracing now shown for a .NET 8 BackgroundService app

### DISCUSSION THREAD

**Discussion by nabeel-servcorp** [datalust/seq-tickets#2141] [Help]
I added Seq tracing capabilities in two projects I am working on. One is ASP.NET Core Web API project in which it is working fine and I can see the traces in Seq. For the other project which is a .NET  8 `BackgroundService`, I can't see the traces even though I see log entries with "_SQL SELECT Bookings_" text. I used the same code in both which I picked from [the official page](https://docs.datalust.co/docs/serilogtracing). I am using **SerilogTracing** nuget. In the `BackgroundService` project, I just removed the `.Instrument.AspNetCoreRequests()` part because it is a background task and not ASP.NET Core app. I am only interested in SQL Server queries instrumentation so I use `.Instrument.SqlClientCommands()`.

I am not sending any explicit spans using `Log.Logger.StartActivity()` method. I just want the built-in traces to appear for SQL queries.

Should I see traces for a `BackgroundService` class?

**Comment by liammclennan** [comment]
Hi @nabeel-servcorp,

You will see traces for you `BackgroundService` project if you send them to Seq. What client are you using?

If you are using OpenTelemetry-dotnet then [this example](https://github.com/datalust/seq-examples/blob/main/client/opentelemetry-csharp-sdk-to-seq/weather/Program.cs) might help.

**Reply by nabeel-servcorp** [reply]
I am using `SerilogTracing` nuget. I used **the same code snippet** in my ASP.NET Core Web API project and it works fine. I updated my question slightly to show that I am using `.Instrument.SqlClientCommands()` SQL queries instrumentation as I did in other project.

**Reply by nblumhardt** [reply]
Hi @nabeel-servcorp; if you wrap (one of) your database calls with `logger.StartActivity()`, does any trace information show up? Thanks!

**Reply by nabeel-servcorp** [reply]
I will try that and update here. I guess the reason might be that when we use `.Instrument.AspNetCoreRequests()` with `.Instrument.SqlClientCommands()` the library implicitly marks every HTTP request as an activity but in a non-web application, this doesn't happen.

**Reply by nabeel-servcorp** [reply]
I tested the suggestion. Still no trace appears.
I wrapped my database calls with this:
`using var activity = Log.Logger.StartActivity("MyBackgroundWorker job");`
and I can see "_MyBackgroundWorker job_" log statement as shown as a plain old log statement (pre-trace era) in Seq.

**Reply by nblumhardt** [reply]
Thanks for checking that out. Are you using version 7.0.0 of Serilog.Sinks.Seq in that project? Earlier versions don't support tracing.

**Reply by nabeel-servcorp** [reply]
Yes, am using version 7.0.0.

**Reply by nblumhardt** [reply]
Interesting! Something unexpected is going on here; I just created a small app:

**Program.cs**

```csharp
using Serilog;
using SerilogTracing;
using worker;

Log.Logger = new LoggerConfiguration()
    .WriteTo.Seq("http://localhost:5341")
    .CreateLogger();

using var _ = new ActivityListenerConfiguration()
    .TraceToSharedLogger();

var builder = Host.CreateApplicationBuilder(args);
builder.Services.AddHostedService<Worker>();

var host = builder.Build();
host.Run();
```

**Worker.cs**

```csharp
using Serilog;
using SerilogTracing;

namespace worker;

public class Worker : BackgroundService
{
    private readonly ILogger<Worker> _logger;

    public Worker(ILogger<Worker> logger)
    {
        _logger = logger;
    }

    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
    {
        while (!stoppingToken.IsCancellationRequested)
        {
            using var activity = Log.Logger.StartActivity("doing some work");
            await Task.Delay(1000, stoppingToken);
        }
    }
}
```

**Worker.csproj**

```xml
<Project Sdk="Microsoft.NET.Sdk.Worker">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <UserSecretsId>dotnet-worker-1c0cd357-f5e8-48f5-85a8-9c6aed4ce259</UserSecretsId>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="8.0.0" />
    <PackageReference Include="serilog.sinks.seq" Version="7.0.0" />
    <PackageReference Include="serilogtracing" Version="1.0.0" />
  </ItemGroup>
</Project>
```

And on .NET 8/Windows, this produces normal traces for me:

![image](https://github.com/datalust/seq-tickets/assets/342712/7e74412c-b702-4386-ae72-b8c56c7fc419)

Can you spot the difference? Or if your configuration is different, can you modify this minimal app until the traces no longer appear correctly? Keen to hear how you go!

**Reply by nabeel-servcorp** [reply]
It is fixed for me too now.
